dashboard/http_server_agent.py

```diff
@@ -53,12 +53,22 @@ async def start(self, modules):
 
         self.runner = aiohttp.web.AppRunner(app)
         await self.runner.setup()
-        site = aiohttp.web.TCPSite(
-            self.runner,
-            "127.0.0.1" if self.ip == "127.0.0.1" else "0.0.0.0",
-            self.listen_port,
-        )
-        await site.start()
+        try:
+            site = aiohttp.web.TCPSite(
+                self.runner,
+                "127.0.0.1" if self.ip == "127.0.0.1" else "0.0.0.0",
+                self.listen_port,
+            )
+            await site.start()
+        except OSError as e:
```

Comment:
try by increasing port by 1 until the port is free

Reply:
In the future, we should add a CLI command or an API that allows the user to tell what port this is after launching their Ray cluster. This is necessary when specifying the dashboard agent address in CLI addresses.

Reply:
Based on @rkooo567 comment, I think by default, we can make it random?

Reply:
Based on @edoakes feed back, if we fail to use the port, we should crash, since KubeRay has no way to access it.